### DISCUSSION TITLE
Question about https://newzone.top/web/VuePress.html#%E5%88%9D%E5%A7%8B%E9%85%8D%E7%BD%AE

### DISCUSSION THREAD

**Discussion by Mister-Hope** [rockbenben/LearnData#13] [General]
我对这页的内容存在比较大的疑问和部分异议，来问问。

> 即使你并没有更新文章，生成的静态文件也会改变。比如我的笔记网站用的 VuePress 默认配置，每次服务器部署需要 10 分钟，期间打开网站就会出错。

这里你是用什么部署的？FTP么？

> 在找到 chainWebpack 配置前，我依照 vue.config.js 添加了 filenameHashing: false，但 VuePress 并未停止 hashname。

另外，这里的 filenameHasing 是 vue-cli 自己加的一个属性。拿到这里当例子很奇怪。就好比你按照 iOS 使用手册去设置 Android 结果无效，本不应该有效。另外本身为了防止应用程序出错，你也从不应该移除hash。

> 使用 vuepress-plugin-seo2 的 ogp 参数对 meta 重新设置，删除不想要的参数。不过这会导致博客的自动摘要功能失效，而且 git 后参数并没发生变化。在 theme.ts 设置 ogp 直接为无效。

这里如果你想要移除，直接 delete 相关属性返回这个对象即可。“自动摘要失效” “git 后参数并没发生变化”是什么情况？如果觉得有问题欢迎开 issue

**Comment by rockbenben** [comment]
> 我对这页的内容存在比较大的疑问和部分异议，来问问。
> 
> > 即使你并没有更新文章，生成的静态文件也会改变。比如我的笔记网站用的 VuePress 默认配置，每次服务器部署需要 10 分钟，期间打开网站就会出错。
> 
> 这里你是用什么部署的？FTP么？

是的，用 SamKirkland/FTP-Deploy-Action@4.3.3 进行 FTP 同步的。最近，更新需要 5-15 分钟。

> > 在找到 chainWebpack 配置前，我依照 vue.config.js 添加了 filenameHashing: false，但 VuePress 并未停止 hashname。
> 
> 另外，这里的 filenameHasing 是 vue-cli 自己加的一个属性。拿到这里当例子很奇怪。就好比你按照 iOS 使用手册去设置 Android 结果无效，本不应该有效。另外本身为了防止应用程序出错，你也从不应该移除hash。

那是我理解错了，已更新备注。

> > 使用 vuepress-plugin-seo2 的 ogp 参数对 meta 重新设置，删除不想要的参数。不过这会导致博客的自动摘要功能失效，而且 git 后参数并没发生变化。在 theme.ts 设置 ogp 直接为无效。
> 
> 这里如果你想要移除，直接 delete 相关属性返回这个对象即可。“自动摘要失效” “git 后参数并没发生变化”是什么情况？如果觉得有问题欢迎开 issue

“自动摘要失效” 可能是当时版本的原因。

“git 后参数并没发生变化” 我也想不起来了，应该是 delete 相关属性失败导致的。我不知道如何移除 opg 参数，当时参考了你给的示例并将其设置为空。 但是，即使值为空，问题仍然存在，之后就删除了相关的设置。

请问，应该如何删除 opg 相关属性？是按照以下方式将其设为 undefined 吗？

```ts
import { seoPlugin } from "vuepress-plugin-seo2";
export default defineUserConfig({
  ...
  plugins: [
    seoPlugin({
      hostname: "https://vuepress-theme-hope.github.io",
      ogp: (ogp, page) => ({
        ...ogp,
        "og:modified_time": "",
      }),
      pageFilter: (page) => ({
        ...page,
        frontmatter: {
          ...page.frontmatter,
          updatedTime: undefined,
        },
      }),
    }),
  ],
});

```

**Reply by Mister-Hope** [reply]
> 请问，应该如何删除 opg 相关属性？是按照以下方式将其设为 undefined 吗？

API的设计很容易理解就是给你个自动生成的对象然后等你返回。

所以你只需要在原对象上把属性删掉，返回这个对象就是了。

**Reply by Mister-Hope** [reply]
> 是的，用 SamKirkland/FTP-Deploy-Action@4.3.3 进行 FTP 同步的。最近，更新需要 5-15 分钟。

不建议用 FTP，你可以酌情考虑用git来进行操作。先接收最新的commit然后将最新的commit释放到网站根目录中。

**Reply by Mister-Hope** [reply]
即使你仍希望使FTP，你也应该先传输assets文件夹的内容，之后传输HTML，通常不会有任何问题。

**Reply by rockbenben** [reply]
> > 是的，用 SamKirkland/FTP-Deploy-Action@4.3.3 进行 FTP 同步的。最近，更新需要 5-15 分钟。
> 
> 不建议用 FTP，你可以酌情考虑用git来进行操作。先接收最新的commit然后将最新的commit释放到网站根目录中。

有没有推荐的 Action？试过几个，只有 SamKirkland/FTP-Deploy-Action@4.3.3 比较稳定
